components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/filter/AuthParameterFilter.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    /**
+     * Cache the parameters retrieved from the API using SessionDataKey.
+     *
+     * @param servletRequest - Received servlet request.
+     * @return - HttpServletRequestWrapper with cached parameter map.
+     */
+    private HttpServletRequest cacheParamsFromSessionDataKey(HttpServletRequest servletRequest) {
+
+        String authAPIURL = buildAPIPath("sessionDataKey", servletRequest.getParameter("sessionDataKey"));
+        String contextProperties = AuthContextAPIClient.getContextProperties(authAPIURL);
+        Gson gson = new Gson();
+        Map<String, Object> cachedParams = gson.fromJson(contextProperties, Map.class);
+
+        return new AuthenticationRequestWrapper(servletRequest, cachedParams);
+    }
+
+    /**
+     * Cache the parameters retrieved from the API using SessionDataKey.
+     *
+     * @param servletRequest - Received servlet request.
+     * @return - HttpServletRequestWrapper with cached parameter map.
+     */
+    private HttpServletRequest cacheParamsFromConsentKey(HttpServletRequest servletRequest) {
+
+        String authAPIURL = buildAPIPath("sessionDataKeyConsent", servletRequest.getParameter("sessionDataKeyConsent"));
+        String contextProperties = AuthContextAPIClient.getContextProperties(authAPIURL);
+        Gson gson = new Gson();
+        Map<String, Object> cachedParams = gson.fromJson(contextProperties, Map.class);
+        return new AuthenticationRequestWrapper(servletRequest, cachedParams);
+    }
+
+    private HttpServletRequest cacheParamsFromErrorKey(HttpServletRequest servletRequest) {
+        String authAPIURL = buildAPIPath("errorKey", servletRequest.getParameter("errorKey"));
+        String contextProperties = AuthContextAPIClient.getContextProperties(authAPIURL);
+        Gson gson = new Gson();
+        Map<String, Object> cachedParams = gson.fromJson(contextProperties, Map.class);
+        return new AuthenticationRequestWrapper(servletRequest, cachedParams);
+    }
+
+    /**
+     * Build the API Path using the key and value.
+     *
+     * @param key   - SessionDataKey or SessionDataKeyConsent.
+     * @param value - Value of the key.
+     * @return - API Path.
+     */
+    private String buildAPIPath(String key, String value) {
+
+        String authAPIURL = context.getInitParameter(Constants.AUTHENTICATION_REST_ENDPOINT_URL);
+
+        if (!StringUtils.isBlank(authAPIURL)) {
+            if (resolveTenantDomain().equals("carbon.super")) {
+                authAPIURL = authAPIURL.replace("t/${tenant}/", "");
+            } else {
+                authAPIURL = authAPIURL.replace("${tenant}", resolveTenantDomain());
+            }
+        } else {
+            try {
+                String contextPath = "/api/identity/auth/v1.1/data";
+                authAPIURL = ServiceURLBuilder.create().addPath(contextPath).build().getAbsolutePublicURL();
+            } catch (URLBuilderException e) {
+                throw new RuntimeException(e);
```

Comment:
How does the login page or error page handle this?
Would be good to throw with a message instead of just the exception